### PR TITLE
Enable more warnings when using gcc.

### PR DESCRIPTION
**Pull request by est77**
- Added APPLESEED_UNUSED to mark unused variables.
- Removed unused variables, unused local typedefs, ...
- Added missing virtual destructor.

### CODE REVIEW

File: src/appleseed/renderer/modeling/project/projectfilewriter.cpp [hunk 1]
```diff
@@ -110,7 +110,6 @@ namespace renderer
 namespace
 {
     // Floating-point formatting settings.
-    const char* VectorFormat     = "%.15f";
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/project/projectfilewriter.cpp]: Surprised about this one. How do we control precision when writing vectors then?

File: src/appleseed/foundation/platform/compiler.h [hunk 2]
```diff
@@ -191,6 +191,17 @@ namespace foundation
 
 
 //
+//  A macro to mark a variable as unused. Useful in unit tests.
```
Comment (dictoon) [inline src/appleseed/foundation/platform/compiler.h]: Cool!

File: src/appleseed.studio/utility/widgetzoomhandler.cpp [hunk 3]
```diff
@@ -227,9 +227,6 @@ void WidgetZoomHandler::apply_scale_factor()
     if (m_scroll_area->widget()->layout())
         m_scroll_area->widget()->layout()->activate();
 
-    const int new_widget_width = m_scroll_area->widget()->width();
-    const int new_widget_height = m_scroll_area->widget()->height();
-
     const double actual_h_multiplier = static_cast<double>(m_scroll_area->widget()->width()) / old_widget_width;
```
Comment (dictoon) [inline src/appleseed.studio/utility/widgetzoomhandler.cpp]: I think the idea here was to use theses variables to compute `actual_h_multiplier`, etc.

File: src/appleseed/foundation/meta/tests/test_dictionary.cpp [hunk 4]
```diff
@@ -110,7 +110,7 @@ TEST_SUITE(Foundation_Utility_StringDictionary)
 
         EXPECT_EXCEPTION(ExceptionDictionaryItemNotFound,
         {
-            const char* item = sd.get("key");
+            APPLESEED_UNUSED const char* item = sd.get("key") ;
```
Comment (dictoon) [inline src/appleseed/foundation/meta/tests/test_dictionary.cpp]: Remove redundant space before semicolon.

File: src/cmake/config/linux-gcc.txt [hunk 5]
```diff
@@ -60,8 +60,13 @@ set (git_command "git")
 
 # Flags common to all configurations.
 set (c_compiler_flags_common
-    -Werror                                             # Treat Warnings As Errors
+    -Wall
```
Comment (dictoon) [inline src/cmake/config/linux-gcc.txt]: Thanks for that, very useful.

File: src/appleseed/renderer/modeling/project/projectfilewriter.cpp (hunk 1 above)
Comment (est77) [inline src/appleseed/renderer/modeling/project/projectfilewriter.cpp]: Could it be that we are not writting any vector?

Comment (dictoon) [inline src/appleseed/renderer/modeling/project/projectfilewriter.cpp]: You're right, we don't. The vectors I had in mind were rotation axes for `<rotation>` elements and scaling factors for `<scaling>` elements, but these are collapsed to matrices when writing project files.

**Pull request merged**